src/forms/button/ButtonGroup.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import {RadioButton} from "../radio";
+import * as utils from "../../utils";
+import {ReactElement, ReactNode} from "react";
+import {classNames} from "../../utils";
+
+type ButtonGroupProps = {
+    className?: string;
+    children?: React.ReactNode[];
+    style?: any;
+    defaultValue?: any;
+    label?: string;
+    disabled?: boolean;
+    isRadio?: boolean;
+    name: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+    onClick?: (evt: React.ClickEvent<HTMLInputElement>) => void;
+};
+
+const initialState = {value: null};
+
+type ButtonGroupState = Readonly<typeof initialState>;
+
+export class ButtonGroup extends React.PureComponent<ButtonGroupProps> {
+    readonly state: ButtonGroupState = initialState;
+
+    constructor(props: ButtonGroupProps) {
+        super(props);
+        const {defaultValue} = props;
+        if (defaultValue) this.state = {value: defaultValue};
+    }
+
+    private handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+        this.setState({value: evt.target.value});
+        const {onChange} = this.props;
+        if (onChange) onChange(evt);
+    };
+
+    private handleClick = (evt: React.ClickEvent<HTMLInputElement>) => {
```

Comment:
should be able to remove this per comment above.

Reply:
Removed the onChange( ) as per previous comment.